### ISSUE TITLE
Co se děje při submitu formuláře

### ISSUE THREAD

**Issue by janmottl** [nette/docs#646]
v dokumentaci mi chybí popis toho co se děje při submitu formuláře. Koukal jsem do životního cyklu presenteru a nemohl jsem tam najít co jsem potřeboval vědět. Chvíli mi trvalo než mi došlo, že se při sumbitu vytvoří formulář nezávisle na renderu a při get se vytvoří formulář během renderování. Hodilo by se doplnění formuláře do životního cyklu presenteru nebo zvlášni životní cyklus formuláře,

Kvituji vylepšení dokumentace za cca poslední rok. Za dobu co dělám s Nette tak se potěšeně zlepšila.

Honza

**Comment by f3l1x** [comment]
To nas moc tesi. 

-----

Musim se priznat, ze jsem to z toho popisku uplne nepochopil. Co by podle tebe melo byt v dokumentaci?

**Comment by janmottl** [comment]
Dělám s weby a Nette krátce. Chybí mi tam popis co se děje při POSTu. Domnívám se, že při POSTu je to nějak takto:
  startup
  action<Action>
  handle<Signal>
  createComponent<formComponent>
  onSubmit
  onSuccess nebo onError (redirect nebo redraw je na aplikaci a spousti se vetev before render, render... 
  shutdown

btw pri redraw se reateComponent<formComponent> nevola, pokud jiz komponenta existuje. To mi pak dela problemy kdych chci aktualizovat obsah formulare. To bych se taky rad docetl jak na to :) https://forum.nette.org/cs/29294-znovunacteni-dat-formulare-po-ajax-submitu

**Comment by janmottl** [comment]
sezralo mi to znaky mensi a vetsi a to co bylo mezi nimi. createComponent minim komponentu formulare.

**Comment by f3l1x** [comment]
To flow co jsi popsal je spravne, akorat si nejsem jisty, jestli by to melo byt v popisu formulare. Resp. dost toho je obsazeno v [zivotnim cyklu presenteru](https://doc.nette.org/cs/2.4/presenters#toc-zivotni-cyklus-presenteru). Myslis, ze stoji za to, to davat jeste nekam jinam?

**Comment by janmottl** [comment]
Ano, do zivotniho cyklu presenteru by se to hodilo. Taky jsem to tam hledal, kdyz jsem pred vice nez rokem dumal nad tim, proc mi fomular nefunguje a createComponent  se vola pred render. V zivotnim cyklu by se hodilo znazorneni kdy se vola createComponent a mozna v textu naznacit na ktere komponenty:
- v pripade POST se zrejme createComponent vola pouze na postovanou komponentu. 
- po render se createComponent vola podle sablony. V pripade snippetu tak na nezijici komponenty v invalidovanych snippetech. Pokud se po postu zmenila db data nad ramec postovanych dat,, tak je nutne nastavit formulari znovu hodnoty interni metodou setValues(). Pokud se zmenila struktura formulare, je treba komponentu znovu vytvorit, napr. removeComponent() po ulozeni formulare.

**Comment by mildabre** [comment]
Ahoj,

dnes jsem si také všiml, že srozumitelnější popis jak formuláře pracují - takový, aby Nette nováček pochopil jak to vlastně funguje by byl potřeba. On to nakonec pokusy a omyly každý rozlouskne, ale zbytečně se na to spálí mnoho času. Zkusím časem nějaký popis připravit a dát do PR.
